### ISSUE TITLE
Add e2e tests for APM / Kibana association

### ISSUE THREAD

**Issue by barkbay** [elastic/cloud-on-k8s#3154]
Following-up #3043 we should add some end to end tests to validate that an APM Server is correctly associated with Kibana.

**Comment by barkbay** [comment]
I think we can implement the following scenario:

1. Create an agent configuration using the Kibana API, the API [is officially documented](https://www.elastic.co/guide/en/kibana/7.7/agent-config-api.html#apm-update-config) from 7.7.0 but is available since 7.5.0

    `PUT https://x.x.x.x:5601/api/apm/settings/agent-configuration`

    The following HTTP body creates the default agent configuration _(a.k.a. `All`)_ :

    ```json
    {
        "service": {},
        "settings": {
            "transaction_sample_rate": 1,
            "capture_body": "errors",
            "transaction_max_spans": 100
        }
    }
    ```

2. Fetch the configuration from the APM Server with the [Agent configuration API](https://www.elastic.co/guide/en/apm/server/current/agent-configuration-api.html)

    `GET https://y.y.y.y:8200/config/v1/agents?service.name=All`
    
    Compare the results:
    
    ````json
    {
      "capture_body": "errors",
      "transaction_max_spans": "100",
      "transaction_sample_rate": "1"
    }
    ````

**Comment by pebrc** [comment]
Very good idea!
